plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java

```diff
@@ -4941,6 +4941,71 @@ public void testPreferredInsertLayout()
         }
     }
 
+    @Test
+    public void testInsertBucketedTableLayout()
+            throws Exception
+    {
+        SchemaTableName tableName = temporaryTable("empty_bucketed_table");
+        try {
+            List<Column> columns = ImmutableList.of(
+                    new Column("column1", HIVE_STRING, Optional.empty()),
+                    new Column("column2", HIVE_LONG, Optional.empty()));
+            HiveBucketProperty bucketProperty = new HiveBucketProperty(ImmutableList.of("column1"), BUCKETING_V1, 4, ImmutableList.of());
+            createEmptyTable(tableName, ORC, columns, ImmutableList.of(), Optional.of(bucketProperty));
+
+            try (Transaction transaction = newTransaction()) {
+                ConnectorMetadata metadata = transaction.getMetadata();
+                ConnectorSession session = newSession();
+                ConnectorTableHandle tableHandle = getTableHandle(metadata, tableName);
+                Optional<ConnectorNewTableLayout> insertLayout = metadata.getInsertLayout(session, tableHandle);
+                assertTrue(insertLayout.isPresent());
+                assertEquals(insertLayout.get().getPartitioning(), Optional.of(new HivePartitioningHandle(
+                        bucketProperty.getBucketingVersion(),
+                        bucketProperty.getBucketCount(),
+                        ImmutableList.of(HIVE_STRING),
+                        OptionalInt.empty(),
+                        false)));
+                assertEquals(insertLayout.get().getPartitionColumns(), ImmutableList.of("column1"));
```

Comment:
is it possible to test `getBucketNodeMap` too?

Reply:
added assertions on result on getBucketNodeMap as well